src/lib/dcob/octoclient.rb

```diff
@@ -6,8 +6,8 @@ def self.hookit(repository, callback_url)
       new.hookit(repository, callback_url)
     end
 
-    def self.apply_commit_statuses(repository_id, pr_number)
-      new.apply_commit_statuses(repository_id, pr_number)
+    def self.apply_commit_statuses(repository_id, pr_number, metadata: nil, prometheus: nil)
+      new(prometheus).apply_commit_statuses(repository_id, pr_number, metadata)
```

Comment:
Oy. I hate that I painted you into a corner where you had to change method signatures for this.

I suspect it would be better if we remove the convenience class methods and require the use of ...

```ruby
client = Dcob::Octoclient.new(prometheus, metadata)
client.apply_commit_statuses(repository_id, pr_number)
```